### ISSUE TITLE
Re-use a table after a JOIN

### ISSUE THREAD

**Issue by Bengreen** [confluentinc/ksql#2042]
I have created a table with CTAS, but the table created is not keyed.
```
Name                 : CTTV_T
Type                 : TABLE
Key field            :
Key format           : STRING
Timestamp field      : Not set - using <ROWTIME>
Value format         : JSON
Kafka topic          : CTTV_T (partitions: 4, replication: 1)
```
I need the table to be keyed so that I can use it in a subsequent table join.

I cannot add a `WITH (KEY='uuid')` as this is not available in CTAS.

I have tried to use `GROUP BY uuid` but this fails as `Field UUID is ambiguous` because it is the join criteria that I am using to create the table in the first place ie. `ON ct.uuid = tv.uuid`.
So I tried to be specific by changing to `GROUP BY ct.uuid`. But that is not on the select statement so not allowed.

How should I progress to create a table using CTAS with a correctly built key.

**Comment by apurvam** [comment]
I think you are hitting https://github.com/confluentinc/ksql/issues/1550, and is fixed by https://github.com/confluentinc/ksql/pull/1697

The fix will be part of the 5.0.1 release, ETA sometime this October.

**Comment by big-andy-coates** [comment]
There is also a lot more work going on for improving key fields. Latest one being https://github.com/confluentinc/ksql/pull/2735.  These will be released in 5.3

Closing this issue as it looks to be resolved.
